src/lib/utils/options/sources/component.ts

```diff
@@ -33,24 +33,6 @@ export class ComponentSource extends OptionsComponent {
             }
         }
     }
-
-    private removeComponent(component: AbstractComponent<any>) {
```

Comment:
Why was this method removed?

Reply:
It was breaking the build as it was defined, but never used: from #641:

```text
src/lib/utils/options/sources/component.ts(37,13): error TS6133: 'removeComponent' is declared but its value is never read.
```

Reply:
Ah, I see. I probably should review this class to ensure that there is not an implementation bug here before I merge the PR.

Reply:
Looks like `removeComponent` should have been called in `onComponentRemoved`. I probably should put together a separate PR and do some testing. Looks like the minimal theme triggers removing components.

https://github.com/TypeStrong/typedoc/blob/6edcbd53e0c85a063bb91cb6732ced7c21e44007/src/lib/output/themes/MinimalTheme.ts#L21-L24

There also looks to be some bugs in `removeComponent` (i.e. `slice` on this array will do nothing).

https://github.com/TypeStrong/typedoc/blob/6edcbd53e0c85a063bb91cb6732ced7c21e44007/src/lib/utils/options/sources/component.ts#L40-L42